tools/mzxmlshaper/mzXMLShaper.xml

```diff
@@ -0,0 +1,152 @@
+<tool id="mzxmlshaper" name="Mz(X)ML Shaper" version="1.0.0+galaxy0" profile="21.01">
+  <description>
+    enables the conversion of open format files into XCMS-supported mz(X)ML file formats
+  </description>
+  <requirements>
+    <requirement type="package" version="4.3.3">r-base</requirement>
+    <requirement type="package" version="0.42.0">bioconductor-msdata</requirement>
+    <requirement type="package" version="2.36.0">bioconductor-mzr</requirement>
+    <requirement type="package" version="1.0.0">r-w4mrutils</requirement>
+  </requirements>
+  <command detect_errors="aggressive"><![CDATA[
+    Rscript '$__tool_directory__/mzXMLShaper.R'
+      inputfilename '$infile'
+      outputfileformat '$outformat'
+      #if str($outformat) == 'mzml':
+        outputfilename '$mzml_file'
+      #end if
+      #if str($outformat) == 'mzXml':
+        outputfilename '$mzXml_file'
+      #end if
+  ]]></command>
+  <inputs>
+    <param name="infile" type="data" format="mzml,mzxml,netcdf" label="File(s) to convert" help="file or collection"/>
+    <param name="outformat" type="select" label="Desired output format">
+      <option value="mzml">mzML</option>
+      <option value="mzXml">mzXML</option>
+    </param>
+  </inputs>
+  <outputs>
+    <data format="mzml" label="${infile.name.rsplit('.',1)[0]}.mzml" name="mzml_file">
+      <filter> outformat=="mzml"</filter>
+    </data>
+    <data format="mzxml" label="${infile.name.rsplit('.',1)[0]}.mzXml" name="mzXml_file">
+      <filter> outformat=="mzXml"</filter>
+    </data>
+  </outputs>
+  <tests>
+    <test expect_num_outputs="1">
+      <param name="infile" value="original_data/111-1_POS01.CDF"/>
+      <param name="outformat" value="mzml"/>
+      <output name="mzml_file" file="conversion_result_cdf_to_mzml.mzml" />
```

Comment:
you could use asserts in your tests, if you don't want to save entire outputs. https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents

Reply:
Resolved in [8615bab](https://github.com/workflow4metabolomics/tools-metabolomics/pull/273/commits/8615bab50e90a4eeebfb713d6e6e9128e5a5bce5). I uploaded smaller files (~3Mb each), but to check that the conversion is successful they have to be tested in their entirety. However it might be impossible to use files smaller than 1Mb to pass the file size test.